### PR TITLE
Sketching out NB for the ZTP roles

### PR DESCRIPTION
**Pull request by tomikazi**

### CODE REVIEW

File: pkg/northbound/proto/roles.proto [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+limitations under the License.
+*/
+
+syntax = "proto3";
+
+package proto;
+
+// DeviceRoleConfig carries the template configuration associated with a device role
+message DeviceRoleConfig {
+    string role = 1;
```
Comment (SeanCondon) [inline pkg/northbound/proto/roles.proto]: probably too early to make this an enumerated list

File: pkg/northbound/proto/roles.proto [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+// DeviceRoleRequest is a request for device role configuration.
+message DeviceRoleRequest {
+    string role = 1;
+}
+
+// DeviceRoleService provides means for setting up device role configurations
+// in support of zero-touch provisioning activities.
+service DeviceRoleService {
```
Comment (SeanCondon) [inline pkg/northbound/proto/roles.proto]: the similarity of this to gNMI (with the CRUD operations) make me wonder if we should be defining this as a YANG model instead and treating it like a config - not of a device but of onos-config itself - a self management model

File: pkg/northbound/proto/roles.proto (hunk 1 above)
Comment (tomikazi) [reply pkg/northbound/proto/roles.proto]: Yes. Also, the idea is that operators may have notion or arbitrary roles, so this needs to stay open-ended.

File: pkg/northbound/proto/roles.proto (hunk 2 above)
Comment (tomikazi) [reply pkg/northbound/proto/roles.proto]: This has crossed my mind as well, but given the fact that we may need to incorporate not just gNMI-based configs, but also P4Runtime based ones and possible whatever the RAN subsystem may add on beyond these two, I thought we ought to for now keep it separate and its own thing.  I purposely chose this form of API to vet-out Carmelo's suggestion of using this style of API for the topology subsystem.

Comment (SeanCondon) [reply pkg/northbound/proto/roles.proto]: sounds logical

**Pull request merged**